src/SFA.DAS.EmployerAccounts/Data/EmployerAccountRepository.cs

```diff
@@ -82,56 +82,48 @@ public async Task<Account> GetAccountByHashedId(string hashedAccountId)
         }
 
         public async Task<AccountDetail> GetAccountDetailByHashedId(string hashedAccountId)
-        {        
-            var accountDetail = await _db.Value.Accounts
-                .Where(ac => ac.HashedId == hashedAccountId)
-                .Select(ac => new AccountDetail
-                {
-                    AccountId = ac.Id,
-                    HashedId = ac.HashedId,
-                    PublicHashedId = ac.PublicHashedId,
-                    Name = ac.Name,
-                    CreatedDate = ac.CreatedDate,
-                    ApprenticeshipEmployerType = (ApprenticeshipEmployerType) ac.ApprenticeshipEmployerType
-                }).FirstOrDefaultAsync();
-
-            if (accountDetail == null)
-            {             
+        {
+            var account = await _db.Value.Accounts.Include(x => x.AccountLegalEntities.Select(y => y.Agreements)).SingleOrDefaultAsync(x => x.HashedId == hashedAccountId);
+
+            if (account == null)
+            {
                 return null;
             }
 
-            accountDetail.OwnerEmail = await _db.Value.Memberships
+            var accountDetail = new AccountDetail
+            {
+                AccountId = account.Id,
+                HashedId = account.HashedId,
+                PublicHashedId = account.PublicHashedId,
+                Name = account.Name,
+                CreatedDate = account.CreatedDate,
+                ApprenticeshipEmployerType = (ApprenticeshipEmployerType)account.ApprenticeshipEmployerType
+            };
+
+            var activeLegalEntities = account.AccountLegalEntities.Where(x =>
+                x.Deleted == null && x.Agreements.Any(ea =>
+                    ea.StatusId == EmployerAgreementStatus.Pending ||
+                    ea.StatusId == EmployerAgreementStatus.Signed));
+
+            accountDetail.LegalEntities = activeLegalEntities.Select(x => x.Id).ToList();
+            accountDetail.AccountAgreementTypes = account.AccountLegalEntities.SelectMany(x => x.Agreements).Select(x => x.Template.AgreementType).Distinct().ToList();
+
+            var ownerEmailTask = _db.Value.Memberships
```

Comment:
All looks good to me in terms of the query as it mimics what was there before just optimised. Only comment is that given that both Account History, and Memberships we are querying by accountId, is it worth modifying the entity models to reflect these relationships? which would mean you could include account history and membership in your top query, do one join up front and then you will have all the information you need?

Reply:
I did consider it but bottled making the change!  Believe it or not, these changes were meant to be as light touch as possible. 😂